### PR TITLE
Migrate erc1155 tests

### LINKED ISSUE #1010: Migrate ERC1155 tests

### PR DESCRIPTION
**Pull request by ericnordelo**
Fixes #1010

### CODE REVIEW

**Review by andrew-fleming — COMMENTED** [review]
Looking good! I left a couple small comments. We also need to settle whether we're adding comments or not on `ignore`s https://github.com/OpenZeppelin/cairo-contracts/pull/1027#discussion_r1661714103

File: src/tests/token/erc1155/common.cairo [hunk 1]
```diff
@@ -1,134 +1,36 @@
...
+    utils::declare_and_deploy("SnakeERC1155ReceiverMock", array![])
 }
 
 pub(crate) fn setup_camel_receiver() -> ContractAddress {
-    utils::deploy(CamelERC1155ReceiverMock::TEST_CLASS_HASH, array![])
+    utils::declare_and_deploy("CamelERC1155ReceiverMock", array![])
 }
 
 pub(crate) fn setup_account() -> ContractAddress {
     let mut calldata = array![PUBKEY];
```
Comment (immrsd) [inline src/tests/token/erc1155/common.cairo:21]:
```suggestion
    let calldata = array![PUBKEY];
```

File: src/tests/token/erc1155/common.cairo [hunk 2]
```diff
@@ -1,134 +1,36 @@
...
-    utils::deploy(SnakeAccountMock::TEST_CLASS_HASH, calldata)
+    utils::declare_and_deploy("SnakeAccountMock", calldata)
 }
 
-pub(crate) fn setup_account_with_salt(salt: felt252) -> ContractAddress {
+pub(crate) fn deploy_another_account_at(
+    existing: ContractAddress, target_address: ContractAddress
+) {
     let mut calldata = array![PUBKEY];
```
Comment (immrsd) [inline src/tests/token/erc1155/common.cairo:28]:
```suggestion
    let calldata = array![PUBKEY];
```

File: src/tests/utils/common.cairo [hunk 3]
```diff
@@ -17,18 +18,27 @@ pub fn deploy_at(
...
 }
 
+/// Deploys a contract from the class hash of another contract which is already deployed.
+pub fn deploy_another_at(
+    existing: ContractAddress, target_address: ContractAddress, calldata: Array<felt252>
+) {
+    let class_hash = get_class_hash(existing);
+    let contract_class = ContractClassTrait::new(class_hash);
+    deploy_at(contract_class, target_address, calldata)
+}
```
Comment (immrsd) [inline src/tests/utils/common.cairo:28]: This helper function will be helpful, it nearly solves the issue of not having `declare_if_not`

**Review by immrsd — APPROVED** [review]
Looks good

**Comment by ericnordelo** [comment]
Let's get this PR merged without the ignored test comments, and I will go over every ignored test (including ERC20 and ERC721 ones) on a separate PR, to fix the ones that can be fixed (safe dispatchers for ex.), and add a comment on the ones that can't be fixed yet.

**Pull request merged**